APMrover2/mode.cpp

```diff
@@ -97,8 +97,13 @@ void Mode::get_pilot_desired_steering_and_throttle(float &steering_out, float &t
 // set desired location
 void Mode::set_desired_location(const struct Location& destination, float next_leg_bearing_cd)
 {
-    // record targets
-    _origin = rover.current_loc;
+    // set origin to last destination if waypoint controller active
+    if ((AP_HAL::millis() - last_steer_to_wp_ms < 100) && _reached_destination) {
```

Comment:
why this timeout ?

Reply:
@khancyr, we need the timeout because the _reached_destination can be left as true in some cases (I think).  So for example, if the vehicle RTLs and then the user switches the vehicle to manual and drives around a bit and then switches to Auto, I think initially the _destination would be home and the _reached_destination would be true.. but that could be from minutes earlier.